File: companion/spss/r/b07.R
veri <- read.csv("companion/spss/csv/b07.csv")
stopifnot(nrow(veri) == 395, ncol(veri) == 10,
          !anyNA(veri),
          all(veri$pass10 == as.integer(veri$g3 >= 10)))
gozlem_sayisi <- nrow(veri)
ortalama <- mean(veri$g3)
standart_sapma <- sd(veri$g3)
oran <- mean(veri$pass10)
frekans <- table(veri$pass10)
print(frekans); print(100 * prop.table(frekans))
print(c(n=gozlem_sayisi, ort=ortalama, ss=standart_sapma,
  sh_ort=standart_sapma/sqrt(gozlem_sayisi), oran=oran,
  sh_oran=sqrt(oran*(1-oran)/gozlem_sayisi)))
print(sum(veri$g3 == 0))

File: companion/spss/python/b07.py
import pandas as pd
import numpy as np

veri = pd.read_csv("companion/spss/csv/b07.csv")
assert veri.shape == (395, 10)
assert not veri.isna().any().any()
assert veri.pass10.eq((veri.g3 >= 10).astype(int)).all()
gozlem_sayisi = len(veri)
ortalama = veri.g3.mean()
standart_sapma = veri.g3.std(ddof=1)
oran = veri.pass10.mean()
frekans = veri.pass10.value_counts().sort_index()
print(frekans); print(100 * frekans / gozlem_sayisi)
print(pd.Series({
    "n": gozlem_sayisi, "ort": ortalama, "ss": standart_sapma,
    "sh_ort": standart_sapma / np.sqrt(gozlem_sayisi),
    "oran": oran,
    "sh_oran": np.sqrt(oran*(1-oran)/gozlem_sayisi)
}))
print(int((veri.g3 == 0).sum()))